### ISSUE TITLE
524 errors are not properly retried in up42-py

### ISSUE THREAD

**Issue by klaucos** [up42/up42-py#596]
We have encountered a 524 error when running a UP42 search which lead to failure of our pipeline. 
It seems like the up42 client retrying was not handling this 524 error.

Our up42-py version == 1.0.0
Executed code: 
```
catalog = up42.initialize_catalog()
search_results = catalog.search(search_parameters, as_dataframe=False)
```
Traceback:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/up42/auth.py", line 122, in request
    response: requests.Response = self._request_helper(request_type, url, data)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/up42/auth.py", line 91, in _request_helper
    response: requests.Response = self.session.request(
                                  ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/up42/http/session.py", line 18, in request
    response.raise_for_status()
  File "/usr/local/lib/python3.11/dist-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 524 Server Error:  for url: https://api.up42.com/catalog/hosts/oneatlas/stac/search

The above exception was the direct cause of the following exception:
***
    search_results = catalog.search(search_parameters, as_dataframe=False)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/up42/catalog.py", line 365, in search
    response_json: dict = self.auth.request("POST", url, search_parameters)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/up42/auth.py", line 144, in request
    raise requests.exceptions.HTTPError(err_message) from err
requests.exceptions.HTTPError: error code: 524
```

Is it possible to improve retrying so that 524 errors are properly retried in up42 client?
Also is there a way to increase the number of retries the up42 client does?

**Comment by javidq** [comment]
Hi @klaucos, due to [the resilience settings](https://github.com/up42/up42-py/blob/e08f93d0f7070bf86e764bc031628aff0e080494/up42/http/config.py#L6) the client retries 3 times with backoff on all server error status codes. I'm not sure that retrying more will necessarily improve the resilience if an endpoint consistently fails. 
`Also is there a way to increase the number of retries the up42 client does?` - we don't think this needs to be part of the client's UX. You can still find ways around since it's Python, but I'd not recommend that.

**Comment by klaucos** [comment]
Hello @javidq  
I am sorry for the confusion, I confirmed that "524" value is really present in request Retry object in status_forcelist attached to session adapter, so retrying should be done.

However we would still prefer if there was a way to increase the number or retries while creating the UP42 client. 
As I said we have encountered and 524 error and we would prefer if we could rather increase number of retries or increase backoff_factor to avoid hitting transient errors while communicating with up42.
